frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -57,6 +57,24 @@ interface NewRunParametersProps {
   handleParameterChange?: (parameters: RuntimeParameters) => void;
 }
 
+const inputConverter = (paramStr: string, paramTypeIdx: number) => {
+  if (ParameterType_ParameterTypeEnum[paramTypeIdx] === 'BOOLEAN' && paramStr === 'True') {
+    return true;
+  } else if (ParameterType_ParameterTypeEnum[paramTypeIdx] === 'BOOLEAN' && paramStr === 'False') {
+    return false;
+  } else if (
+    ParameterType_ParameterTypeEnum[paramTypeIdx] === 'NUMBER_INTEGER' &&
+    Number(paramStr)
```

Comment:
We shouldn't use the current if condition `Number(paramStr)` for two reasons:

1. `paramStr` value can be float type. And `Number(paramStr)` will generate a float type number if input has decimal point. However, users are required to provide an integer based on parameter type. Even though the Protobuf.Value accepts `double` type for any number, we as frontend should do the due diligence for checking user input value is integer.
2. If `paramStr` value is '0', and '0' is a valid number. The result of `Number(paramStr)` actually translates to `false`, we should avoid this special case to be treated as non-number.

Reply:
Great call! I will then use `parseInt()` to process integer-type input.

Reply:
I found that parseInt() will return first integer, which result in wrong output such as `120a` (will return 120), `43 9` (will return 43). Thus, I decide to use double `bitwise negation` method.